lib/buffer-api.js

```diff
@@ -36,9 +36,14 @@ var util = require('util'),
 
           if (err) {
             return reject(err)
-          } else if (res.statusCode !== 200) {
+          } else if (res.statusCode !== 200 || (body && typeof(body.success) !== "undefined" && !body.success)) {
```

Comment:
Can you extract the new condition into a named function?

For example:

```
function hasInvalidResponse(body) {
...
}
```

Also first condition could be extracted as:

```
function isHttpOkCode() {
...
}
```

More easy to maintain it in the future 😄